enterprise/suppress-user/internal/badgerdb/badgerdb.go

```diff
@@ -70,7 +75,7 @@ func NewRepository(basePath string, log logger.Logger, opts ...Opt) (*Repository
 	for _, opt := range opts {
 		opt(b)
 	}
-
+	b.stats = stats.NewStats(config.Default, logger.Default, svcMetric.Instance)
```

Comment:
Can we pass stats as a function parameter, similar to what we are doing for `logger.Logger`?